make tests fails with ENABLE_DATA_TOOLS=Off

Some of the files included in [`test/CMakeLists.txt`](https://github.com/valhalla/valhalla/blob/master/test/CMakeLists.txt#L8) need to have [`ENABLE_DATA_TOOLS`](https://github.com/valhalla/valhalla/blob/master/CMakeLists.txt#L21) option turned on, else the tests fail to build:

Eg:
```shell
$ cd build
$ cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -DLOGGING_LEVEL=WARN -DENABLE_NODE_BINDINGS=Off -DENABLE_PYTHON_BINDINGS=Off -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DENABLE_DATA_TOOLS=Off
$ make tests
...
[ 64%] Linking CXX executable complexrestriction
/usr/bin/ld: CMakeFiles/complexrestriction.dir/complexrestriction.cc.o: in function `(anonymous namespace)::ComplexRestriction_WalkViasBuilder_Test::TestBody()':
/home/mandeep/work/valhalla/test/complexrestriction.cc:26: undefined reference to `valhalla::mjolnir::ComplexRestrictionBuilder::set_via_list(std::vector<valhalla::baldr::GraphId, std::allocator<valhalla::baldr::GraphId> > const&)'
/usr/bin/ld: CMakeFiles/complexrestriction.dir/complexrestriction.cc.o: in function `(anonymous namespace)::ComplexRestriction_WriteRead_Test::TestBody()':
/home/mandeep/work/valhalla/test/complexrestriction.cc:54: undefined reference to `valhalla::mjolnir::ComplexRestrictionBuilder::set_via_list(std::vector<valhalla::baldr::GraphId, std::allocator<valhalla::baldr::GraphId> > const&)'
/usr/bin/ld: /home/mandeep/work/valhalla/test/complexrestriction.cc:57: undefined reference to `valhalla::mjolnir::ComplexRestrictionBuilder::set_via_list(std::vector<valhalla::baldr::GraphId, std::allocator<valhalla::baldr::GraphId> > const&)'
collect2: error: ld returned 1 exit status
make[3]: *** [test/CMakeFiles/complexrestriction.dir/build.make:92: test/complexrestriction] Error 1
make[2]: *** [CMakeFiles/Makefile2:4475: test/CMakeFiles/complexrestriction.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```

and just `make` also fails:
```shell
$ make
...
[ 95%] Linking CXX static library libbenchmark_main.a
[ 95%] Built target benchmark_main
make[2]: *** No rule to make target '../test/valhalla_build_tiles', needed by 'test/data/utrecht_tiles/0/003/196.gph'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:5211: test/CMakeFiles/utrecht_tiles.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -337,7 +337,7 @@ if(ENABLE_TESTS)
 endif()
 
 # NOTE(mookerji): Windows CI seems to break on the gbench build, so shelve Win32 support for now.
-if(ENABLE_BENCHMARKS AND NOT WIN32)
+if(ENABLE_BENCHMARKS AND ENABLE_DATA_TOOLS AND NOT WIN32)
   add_subdirectory(bench)
 endif()
 

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -5,19 +5,19 @@ set_target_properties(gmock PROPERTIES FOLDER "Dependencies")
 set_target_properties(gmock_main PROPERTIES FOLDER "Dependencies")
 
 ## Lists tests
-set(tests aabb2 access_restriction actor admin attributes_controller astar_bss complexrestriction countryaccess datetime directededge
+set(tests aabb2 access_restriction actor admin attributes_controller datetime directededge
   distanceapproximator double_bucket_queue edgecollapser edgestatus ellipse encode
   enhancedtrippath factory graphid graphtile graphtileheader gridded_data grid_range_query grid_traversal instructions
-  json laneconnectivity linesegment2 location logging maneuversbuilder map_matcher_factory mapmatch mapmatch_config
+  json laneconnectivity linesegment2 location logging maneuversbuilder map_matcher_factory mapmatch_config
   narrative_dictionary nodeinfo nodetransition obb2 openlr optimizer parse_request point2 pointll
   polyline2 predictedspeeds queue routing sample sequence sign signs streetname streetnames streetnames_factory
   streetnames_us streetname_us tilehierarchy tiles transitdeparture transitroute transitschedule
-  transitstop turn turnlanes util_midgard util_skadi util_mjolnir vector2 verbal_text_formatter verbal_text_formatter_us
+  transitstop turn turnlanes util_midgard util_skadi vector2 verbal_text_formatter verbal_text_formatter_us
   verbal_text_formatter_us_co verbal_text_formatter_us_tx viterbi_search compression filesystem traffictile)
 
 if(ENABLE_DATA_TOOLS)
-  list(APPEND tests astar edgeinfobuilder graphbuilder graphparser graphtilebuilder graphreader isochrone predictive_traffic
-    idtable matrix minbb multipoint_routes names node_search reach recover_shortcut refs search servicedays shape_attributes signinfo summary urban thor_worker timedep_paths timeparsing trivial_paths uniquenames utrecht lua)
+  list(APPEND tests astar astar_bss complexrestriction countryaccess edgeinfobuilder graphbuilder graphparser graphtilebuilder graphreader isochrone predictive_traffic
+    idtable mapmatch matrix minbb multipoint_routes names node_search reach recover_shortcut refs search servicedays shape_attributes signinfo summary urban thor_worker timedep_paths timeparsing trivial_paths uniquenames util_mjolnir utrecht lua)
   if(ENABLE_HTTP)
     list(APPEND tests http_tiles)
   endif()

